presto-common/src/main/java/com/facebook/presto/common/type/SqlTimestamp.java

```diff
@@ -15,47 +15,78 @@
 
 import com.fasterxml.jackson.annotation.JsonValue;
 
-import java.time.Instant;
 import java.time.ZoneId;
 import java.time.format.DateTimeFormatter;
 import java.util.Objects;
 import java.util.Optional;
 
 import static com.facebook.presto.common.type.TimeZoneKey.UTC_KEY;
+import static com.facebook.presto.common.type.TimestampMicrosUtils.microsToInstant;
+import static java.lang.Math.floorDiv;
 
 public final class SqlTimestamp
 {
     // This needs to be Locale-independent, Java Time's DateTimeFormatter compatible and should never change, as it defines the external API data format.
```

Comment:
This comment says "should never change, as it defines the external API data format". Do you know what that means and whether this change is an exception?

Reply:
Yes, since we change the external API TIMESTAMP format (e.g. parquet / orc / rcfile), it is inevitable to change this class.